tests/test_reweighting.py

```diff
@@ -0,0 +1,407 @@
+from jax.scipy.stats.norm import logpdf as norm_logpdf
+from scipy.stats import norm
+
+from timemachine.constants import BOLTZ
+from timemachine.datasets import fetch_freesolv
+from timemachine.fe.functional import construct_differentiable_interface_fast
+from timemachine.fe.reweighting import (
+    construct_endpoint_reweighting_estimator,
+    construct_mixture_reweighting_estimator,
+    interpret_as_mixture_potential,
+    one_sided_exp,
+)
+from timemachine.ff import Forcefield
+from timemachine.md.enhanced import get_solvent_phase_system
+
+# TODO: should gaussian test system be moved into a test fixture? or perhaps to the testsystems module?
+
+
+def make_gaussian_testsystem():
+    """normalized/unnormalized 1D Gaussian with a dependence on lambda and params"""
+
+    def annealed_gaussian_def(lam, params):
+        initial_mean, initial_log_sigma = 0.0, 0.0
+        target_mean, target_log_sigma = params
+
+        # lam = 0 -> (mean = 0, stddev = 1)
+        # lam = 1 -> (mean = target_mean, stddev = target_sigma)
+        mean = lam * target_mean - (1 - lam) * initial_mean
+        stddev = np.exp(lam * target_log_sigma + (1 - lam) * initial_log_sigma)
+
+        return mean, stddev
+
+    def sample(lam, params, n_samples):
+        mean, stddev = annealed_gaussian_def(lam, params)
+        return norm.rvs(loc=mean, scale=stddev, size=(n_samples, 1))
+
+    def logpdf(x, lam, params):
+        mean, stddev = annealed_gaussian_def(lam, params)
+        return np.sum(norm_logpdf(x, loc=mean, scale=stddev))
+
+    def u_fxn(x, lam, params):
+        """unnormalized version of -logpdf"""
+        mean, stddev = annealed_gaussian_def(lam, params)
+        return np.sum(0.5 * ((x - mean) / stddev) ** 2)
+
+    def normalized_u_fxn(x, lam, params):
+        return -logpdf(x, lam, params)
+
+    def reduced_free_energy(lam, params):
+        mean, stddev = annealed_gaussian_def(lam, params)
+        log_z = np.log(stddev * np.sqrt(2 * np.pi))
+        return -log_z
+
+    return u_fxn, normalized_u_fxn, sample, reduced_free_energy
+
+
+def assert_estimator_accurate(estimate_delta_f, analytical_delta_f, ref_params, n_random_trials, atol):
+    """for many random parameter sets, assert that the reweighted estimates of
+    delta_f(params) and grad(delta_f)(params) are consistent with analytical_delta_f function
+
+    Parameters
+    ----------
+    estimate_delta_f: Callable[[params], float]
+        estimated free energy difference as a
+        differentiable function of params
+    analytical_delta_f: Callable[[params], float]
+        exact free energy difference as a
+        differentiable function of params
+    ref_params: [2,] array
+        generate random parameters near ref_params
+    n_random_trials: int
+    atol: float
+        absolute tolerance
+    """
+
+    def sample_random_params():
+        mean = ref_params[0] + onp.random.rand()
+        log_sigma = ref_params[1] - onp.random.rand()
```

Comment:
Is the idea for log_sigma that we can avoid dealing with negative values of sigma by doing all of this in log space? (Since `exp(log_sigma)` is strictly positive for all `-inf < log sigma < inf`)